src/mavlink-router/endpoint.cpp

```diff
@@ -792,10 +798,15 @@ int UdpEndpoint::write_msg(const struct buffer *pbuf)
     memcpy(&tx_buf.data[tx_buf.len], pbuf->data, pbuf->len);
     tx_buf.len += pbuf->len;
 
-    if (!_write_scheduled) {
+    int ret = pbuf->len;
```

Comment:
This gets tricky with trying to return sane values from the `write_msg` in case the caller is expecting values that would align with nonbuffered writes.